app/scripts/components/exploration/components/datasets/colorRangeSlider.tsx

```diff
@@ -0,0 +1,270 @@
+    colorMapScale?.max ? colorMapScale.max : max
+  );
+  const range = useRef<HTMLDivElement>(null);
+  const middleMarker = useRef<HTMLDivElement>(null);
+
+  // Convert to percentage
+  const getPercent = useCallback(
+    (value) => Math.round(((value - min) / (max - min)) * 100),
+    [min, max]
+  );
+  //Calculate the range
+  const rangeCalculation = (maxPercent, minPercent) => {
+    if (range.current) {
+      range.current.style.width = `${
+        maxPercent - minPercent >= 100 ? 100 : maxPercent - minPercent
+      }%`;
+      if (middleMarker.current)
+        middleMarker.current.style.left = `${range.current.style.width}%`;
+    }
+    return;
+  };
+  const resetErrorOnSlide = (value, slider) => {
+    if (value > min || value < max) {
+      slider === 'max'
+        ? setInputError({ ...inputError, max: false, lessThanMin: false })
+        : setInputError({ ...inputError, min: false, largerThanMax: false });
+    }
+  };
+
+  const minMaxBuffer = 0.001;
+  const textInputClasses =
+    'flex-1 radius-md height-3 font-size-3xs width-5 border-2px ';
+  const thumbPosition = `position-absolute pointer-events width-card height-0 outline-0`;
+
+  useEffect(() => {
+    let maxValPrevious;
+    let minValPrevious;
+    //checking that there are no current errors with inputs
+    if (Object.values(inputError).every((error) => !error)) {
+      //set the filled range bar on initial load
+      if (
+        colorMapScale &&
+        maxVal != maxValPrevious &&
+        minVal != minValPrevious
+      ) {
+        const minPercent = getPercent(minValRef.current);
+        const maxPercent = getPercent(maxValRef.current);
+
+        rangeCalculation(maxPercent, minPercent);
+
+        if (range.current) range.current.style.left = `${minPercent}%`;
+      } else {
+        //set the filled range bar if change to max slider
+        if (maxVal != maxValPrevious) {
+          maxValPrevious = maxVal;
+          const minPercent = getPercent(minValRef.current);
+          const maxPercent = getPercent(maxVal);
+          rangeCalculation(maxPercent, minPercent);
+        }
+        //set the filled range bar if change to min slider
+        if (minVal != minValPrevious) {
+          minValPrevious = minVal;
+          const minPercent = getPercent(minVal);
+          const maxPercent = getPercent(maxValRef.current);
+          rangeCalculation(maxPercent, minPercent);
+          if (range.current) range.current.style.left = `${minPercent}%`;
+        }
+      }
+    }
+    // determining if there is an initial colorMapeScale or if it is the default min and max
+    if (
+      !colorMapScale ||
+      (colorMapScale.max == max && colorMapScale.min == min)
+    ) {
+      setColorMapScale({ min: minVal, max: maxVal });
+    } else
+      setColorMapScale({
+        min: Number(minValRef.current),
+        max: Number(maxValRef.current)
+      });
+  }, [maxVal, minVal, getPercent, setColorMapScale, inputError, max, min]);
```

Comment:
For this dependency array, I get a `React Hook useEffect has a missing dependency: 'colorMapScale'. Either include it or remove the dependency array.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)`